log/sequencer.go

```diff
@@ -430,6 +430,12 @@ func (s Sequencer) IntegrateBatch(ctx context.Context, tree *trillian.Tree, limi
 			Revision:       uint64(newVersion),
 		}
 
+		if currentRoot.RootHash != nil && newLogRoot.TimestampNanos < currentRoot.TimestampNanos {
```

Comment:
I think `currentRoot.RootHash` can't be nil here (it gets checked at L334 and the signer will refuse to run a batch until after the operator has manually called `InitLog`).

Might be worth updating the timestamp test to `<=` though; I suspect most storage impls will have timestamp as part of the key and would refuse to store a 2nd STH with identical timestamp, but you never know - and in practice generating two STH with identical nanosecond timestamp should never occur legitimately.